### PR TITLE
Generic repository, service and controller

### PR DESCRIPTION
**Pull request by Catarina98**

### CODE REVIEW

File: GrocifyApp.API/Controllers/GenericController.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    [Route("[controller]")]
+    public class GenericRestController<T> : ControllerBase where T : BaseEntity
+    {
+        private IEntitiesService<T> EntitiesService { get; set; }
+        public GenericRestController(IEntitiesService<T> _repository)
+        {
+            EntitiesService = _repository;
+        }
+
+        //[HttpGet]
```
Comment (jessywork) [inline GrocifyApp.API/Controllers/GenericController.cs]: why is this commented?

File: GrocifyApp.API/Controllers/GenericController.cs
Comment (jessywork) [inline GrocifyApp.API/Controllers/GenericController.cs:1]: ola

File: GrocifyApp.API/Program.cs [hunk 2]
```diff
@@ -11,7 +11,8 @@
 builder.Services.AddSwaggerGen();
 
 // Add dependencies
```
Comment (Catarina98) [inline GrocifyApp.API/Program.cs:21]: still missing swagger examples working

File: GrocifyApp.API/Controllers/GenericController.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        /// </summary>
+        /// <response code="200">Success.</response>
+        [HttpGet, AllowAnonymous]
+        public async Task<IActionResult> GetAll()
+        {
+            var entities = await _genericBusiness.GetAll();
+            return Ok(entities);
+        }
+
+        //CREATE Entity
```
Comment (jessywork) [inline GrocifyApp.API/Controllers/GenericController.cs]: Uppercase "Entity"

File: GrocifyApp.API/Controllers/GenericController.cs [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        //GET ALL ENTITIES
+        /// <summary>
+        /// Gets all entities.
+        /// </summary>
+        /// <response code="200">Success.</response>
+        [HttpGet, AllowAnonymous]
+        public async Task<IActionResult> GetAll()
+        {
+            var entities = await _genericBusiness.GetAll();
+            return Ok(entities);
```
Comment (jessywork) [inline GrocifyApp.API/Controllers/GenericController.cs:52]: Enter?

File: GrocifyApp.API/Controllers/GenericController.cs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            var u = _mapper.Map<TEntity>(entity);
+
+            try
+            {
+                await _genericBusiness.Insert(u);
+            }
+            catch (Exception ex)
+            {
+                var errors = new List<string> { ex.Message };
+                return BadRequest(new BadRequestModel { Errors = errors });
```
Comment (jessywork) [inline GrocifyApp.API/Controllers/GenericController.cs:83]: Enter?

File: GrocifyApp.API/Controllers/GenericController.cs [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+            }
+
+            try
+            {
+                await _genericBusiness.Update(u);
+            }
+            catch (Exception ex)
+            {
+                var errors = new List<string> { ex.Message };
+                return BadRequest(new BadRequestModel { Errors = errors });
```
Comment (jessywork) [inline GrocifyApp.API/Controllers/GenericController.cs:144]: Enter?

File: GrocifyApp.API/Controllers/GenericController.cs [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+            }
+
+            return Ok();
+
```
Comment (jessywork) [inline GrocifyApp.API/Controllers/GenericController.cs]: Missing Delete by id

File: GrocifyApp.API/Models/RequestModels/ProductSectionRequestModel.cs [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+{
+    public class ProductSectionRequestModel : BaseEntity
+    {
+        /// <example>Meat</example>
+        [Required]
+        public required string Name { get; set; }
+        /// <example>Chicken Icon</example>
+        [Required]
+        public required string Icon { get; set; }
+    }
```
Comment (jessywork) [inline GrocifyApp.API/Models/RequestModels/ProductSectionRequestModel.cs]: Don't we need the house id?

File: GrocifyApp.API/Models/ResponseModels/ProductSectionResponseModel.cs [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    public class ProductSectionResponseModel
+    {
+        ///<example>Id</example>
+        public Guid Id { get; set; }
+
+        /// <example>Meat</example>
+        public required string Name { get; set; }
+
+        /// <example>Chicken Icon</example>
+        public required string Icon { get; set; }
```
Comment (jessywork) [inline GrocifyApp.API/Models/ResponseModels/ProductSectionResponseModel.cs:12]: Don't we need the house id?

**Review by jessywork — APPROVED** [review]

**Pull request merged**